pkg/config/config.go

```diff
@@ -45,31 +43,21 @@ func TemplateConfig(log *logger.Logger, configSpecData string, configValuesData
 	}
 
 	ApplyValuesToConfig(config, configCtx.ItemValues)
-	configDocWithData, err := marshalConfig(config)
+	configDocWithData, err := util.MarshalIndent(2, config)
 	if err != nil {
 		return "", errors.Wrap(err, "failed to marshal config")
 	}
 
 	builder.AddCtx(configCtx)
 
-	rendered, err := builder.RenderTemplate("config", configDocWithData)
+	rendered, err := builder.RenderTemplate("config", string(configDocWithData))
 	if err != nil {
 		return "", errors.Wrap(err, "failed to render config template")
 	}
 
 	return rendered, nil
 }
 
-func marshalConfig(config *kotsv1beta1.Config) (string, error) {
-	s := json.NewYAMLSerializer(json.DefaultMetaFactory, scheme.Scheme, scheme.Scheme)
```

Comment:
why is or isnt this necessary?

Reply:
it's replaced with the `util.MarshalIndent` function which then calls the yaml render function directly

Reply:
but that doesnt use the kubernetes scheme. what is the difference?

Reply:
The kubernetes yaml marshaller isn't special
It just renders json and converts it to yaml (with gopkg.in/yaml.v2, even)
The util function also renders with that package (well, a newer version of it) but sets the maximum line length to unlimited to avoid wrapping

Reply:
Having 2 different marshallers will be difficult over time. What options do we have to eventually get this back to the built-in?  I understand that it's not special, but now our code is because this one file is handled using a different marshal/unmarhshal library. For simplicity and maintainability moving forward, it would be good to have a path to getting this back to the same library as everything else.

Reply:
Perhaps we could add additional tests that do side-by-side renderings of a range of config files, to ensure that the alternate 'unlimited-width' render path produces exactly the same output as the k8s marshaller.  Once the goyaml project merges in Andrew's PR, and that fix gets included with k8s, that would be our path to using a single library everywhere, but until then, the tests would give us some confidence that the behavior hasn't diverged in the meantime.

Reply:
side-by-side rendering added in #320